lib/client.js

```diff
@@ -633,16 +636,27 @@ MqttClient.prototype.removeOutgoingMessage = function (mid) {
 }
 
 /**
- * reconnect - connect again using the same options
+ * reconnect - connect again using the same options as connect()
+ * `incomingStore` and `outgoingStore` need to be ready before `reconnect()` calling.
  *
  * @returns {MqttClient} this - for chaining
  *
  * @api public
  */
 MqttClient.prototype.reconnect = function () {
```

Comment:
We could add the two stores as an option to `reconnect`.

Reply:
It's a good idea!
If stores are passed, `reconnect()` uses them, otherwise renew stores the same as `connect()`.
If users want to reuse existing stores, hold the stores on the user side. And then, pass them to `reconnect()`. It's intuitive.

Reply:
updated.